app/furniture/marketplace/order.rb

```diff
@@ -52,5 +52,136 @@ def marketplace_name
     def price_total
       [product_total, delivery_fee, tax_total].compact.sum
     end
+
+    def square_client
+      @square_client ||= Square::Client.new(access_token: marketplace.square_access_token, environment: marketplace.square_environment)
+    end
+
+    def send_to_square_seller_dashboard
+      square_create_order_response = create_square_order
+      square_create_payment_response = create_square_order_payment(square_create_order_response.body.order[:id], square_create_order_response.body.order[:total_money][:amount])
+
+      # This data is intended for use in debugging, etc... until we further
+      # the Square integration productize
+      {
+        order_id: square_create_order_response.body.order[:id],
+        payment_id: square_create_payment_response.body.payment[:id]
+      }
+    end
+
+    private def square_idemp_key
+      "#{id}_#{8.times.map { rand(10) }.join}"
```

Comment:
Does Square want this to be per-order impotency or is it per-api request impotency?

Reply:
Per request.